autolysis/__init__.py

```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
+        Out[5]:
+        language  English  Mandarin
+        city
+        London         90        40
+        Shanghai       50       350
+
+    However, if the ``city`` and ``language`` were independent, we would expect::
+
+        In [6]: result['expected']
+        Out[6]:
+        language     English    Mandarin
+        city
+        London     34.339623   95.660377
+        Shanghai  105.660377  294.339623
+
+    `Cramer's V <http://en.wikipedia.org/wiki/Cram%C3%A9r%27s_V>`_
+    tells us whether this difference is significant enough to say that
+    ``language`` and ``city`` are independent::
+
+        In [7]: result['V']
+        Out[7]: 0.54866228139712236
+
+    Any value above 0.5 is a strong indication of a relationship. This says
+    that ``language`` and ``city`` are dependent.
+    '''
+    observed = pd.crosstab(index, column, values, aggfunc='sum').replace('', np.nan).fillna(0)
+    chi2, p, dof, expected_values = chi2_contingency(observed.values,
+                                                     correction=correction)
+    expected = pd.DataFrame(expected_values,
+                            index=observed.index,
+                            columns=observed.columns)
+    n = pd.np.sum(observed.values)
+    nindex, ncolumns = len(observed.index), len(observed.columns)
+    return {
+        'observed': observed,
+        'expected': expected,
+        'chi2': chi2,
+        'p': p,
+        'dof': dof,
+        'V': (chi2 / n / (min(nindex, ncolumns) - 1)) ** 0.5,
+    }
+
+
+def crosstabs(data, columns=None, values=None, correction=False, pairs_top=10000):
+    '''
+    Identifies the strength of relationship between every pair of categorical
+    columns in a DataFrame
+
+    Parameters
+    ----------
+    data : Blaze data
+        A data with at least 2 columns having categorical values.
+    columns : list of column names in data
+        If not specified, uses ``autolyse.types(data)['groups']`` to identify
+        all columns with categorical data.
+    values : str, column name
+        Optional column that contains weights to aggregate by summing up. By
+        default, each row is counted as an observation.
+    correction : boolean
+        If True, and the degrees of freedom is 1, apply Yates' correction for
+        continuity. The effect of the correction is to adjust each observed
+        value by 0.5 towards the corresponding expected value. Defaults to False
+        since Cramer's V (a more useful metric than chi-squared) must be computed
+        without this correction.
+    pairs_top: integer, Pick only top 10000 pairs by default
+    '''
+    if columns is None:
+        columns = types(data)['groups']
+
+    result = AttrDict()
+    parameters = ('p', 'chi2', 'dof', 'V')
+    for index, column in itertools.combinations(columns, 2):
+        agg_col = values if values in data.columns else column
+        agg_func = bz.count(data[agg_col]) if agg_col == column else bz.sum(data[agg_col])
```

Comment:
Duh, please ignore all other line-length comments, Apologies. Look into the docstring thing.